portal-web/docroot/html/taglib/ui/input_localized/page.jsp

```diff
@@ -300,40 +300,52 @@
 				placeholder = placeholder + 'Editor';
 			</c:if>
 
-			Liferay.InputLocalized.register(
-				'<%= namespace + id + HtmlUtil.getAUICompatibleId(fieldSuffix) %>',
-				{
-					<c:if test="<%= Validator.isNotNull(activeLocales) && !activeLocales.isEmpty() %>">
-						activeLocales: <%= JSONFactoryUtil.createJSONArray(activeLocales) %>,
-					</c:if>
-
-					adminMode: <%= adminMode %>,
-					availableLocales: available,
-					boundingBox: '#<%= namespace + id %>PaletteBoundingBox',
-					columns: 20,
-					contentBox: '#<%= namespace + id %>PaletteContentBox',
-					defaultLanguageId: defaultLanguageId,
-
-					<c:if test='<%= type.equals("editor") %>'>
-						editor: window['<%= namespace + HtmlUtil.escapeJS(fieldName) + "Editor" %>'],
-					</c:if>
-
-					fieldPrefix: '<%= fieldPrefix %>',
-					fieldPrefixSeparator: '<%= fieldPrefixSeparator %>',
-					helpMessage: '<%= HtmlUtil.escapeJS(helpMessage) %>',
-					id: '<%= id %>',
-					inputPlaceholder: placeholder,
-					inputBox: '#<%= namespace + id %>BoundingBox',
-					items: availableLanguageIds,
-					itemsError: errorLanguageIds,
-					lazy: <%= !type.equals("editor") %>,
-					name: '<%= HtmlUtil.escapeJS(name) %>',
-					namespace: '<%= namespace %>',
-					selectedLanguageId: '<%= selectedLanguageId %>',
-					toggleSelection: false,
-					translatedLanguages: '<%= StringUtil.merge(languageIds) %>'
-				}
-			);
+			Liferay.Loader.require(
+			[
+				A.config.groups.components.mainModule,
+				A.config.groups.react.mainModule,
+				A.config.groups.state.mainModule,
+			],
```

Comment:
This looks a great idea.

Could we use some JSP conditions to inject the `Liferay.Loader.require` only if we have non-empty `activeLocales` in the tag? I'm trying to preserve old behavior as much as possible, so if the require() call is not needed we shall avoid it.

Reply:
Done! https://github.com/liferay-frontend/liferay-portal/pull/1328/commits/eef6bb985b57e1ab9132ae0be87924ce12044bad